sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/base_node.py

```diff
@@ -303,29 +288,30 @@ def _get_component_id(self) -> Union[str, Component]:
             # If component is remote, return it's asset id
             return self._component.id
         # Otherwise, return the component version or arm id.
-        return self._component
+        res: Union[str, Component] = self._component
+        return res
 
-    def _get_component_name(self):
+    def _get_component_name(self) -> Optional[str]:
         # first use component version/job's display name or name as component name
         # make it unique when pipeline build finished.
         if self._component is None:
             return None
         if isinstance(self._component, str):
             return self._component
-        return self._component.name
+        return str(self._component.name)
```

Comment:
Why is it modified like this? `self._component.name `should be guaranteed to be "str"